### ISSUE TITLE
Bump `anyio` to v4 & `fsspec` to v2024

### ISSUE THREAD

**Issue by der-eismann** [snowflakedb/snowflake-ml-python#126]
Hey everyone, we are using `snowflake-ml-python` from within Apache Airflow. In the [constraints file](https://raw.githubusercontent.com/apache/airflow/constraints-2.8.4/constraints-3.10.txt) for the (already outdated) version 2.8.4 they are defining `anyio==4.3.0` & `fsspec==2024.3.1`, causing us some headache when installing this project.
Any chance these dependencies could be updated or the version requirements being relaxed a bit?

**Comment by sfc-gh-sdas** [comment]
Of course, we will give it a try. If nothing else breaks, will take care upgrading dependencies.

**Comment by lennartvandeguchte** [comment]
We also run into issues with anyio when using Prefect, this requires anyio >= 4.4.0, < 5.0.0

**Comment by sfc-gh-sdas** [comment]
We have upgraded `fsspec`. Looking into `anyio`.

**Comment by sfc-gh-wzhao** [comment]
We have bumped the upper bound of anyio to <5 in the 1.7.4 release. I am closing this issue for now, please re-open if you have any problems.
